### PR TITLE
Remove AttributionCollection from default bundle

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

This modifies the injection strategy used by `AttributionCollection` so that consumers which don't want or need attribution don't pay for this functionality in their bundle.

### CODE REVIEW

**Comment by Abe27342** [comment]
draft as this is a WIP and I need some better names. The basic idea is here though so we should get an idea of bundle gain wins.

**Comment by Abe27342** [comment]
note: ideally we could push the split/append logic to the policy handler as well. that doesn't quite work right now b/c snapshot logic condenses segments with append before saving but doesn't raise the event. But it could be a nice further cleanup on top of this.

File: api-report/merge-tree.api.md [hunk 1]
```diff
@@ -305,6 +320,18 @@ export interface IAttributionCollection<T> {
     splitAt(pos: number): IAttributionCollection<T>;
 }
 
+// @internal @sealed (undocumented)
+export interface IAttributionCollectionSerializer {
+    populateAttributionCollections(segments: Iterable<ISegment>, summary: SerializedAttributionCollection): void;
+    // Warning: (ae-forgotten-export) The symbol "SerializedAttributionCollection" needs to be exported by the entry point index.d.ts
```
Comment (Abe27342) [inline api-report/merge-tree.api.md:326]: this bit isn't great. i'd love to not expose the serialized type, but if we want the user to import and specify the method of doing so (even if we implement that factory as is done in this PR), dropping the type here does lose some typesafety. If we did expose it it'd obviously be marked internal.

**Comment by Abe27342** [comment]
marking this as ready for review pending bundle size confirmation that this reduces as expected (if it doesn't, I have more to look into :P)

**Review by clarenceli-msft — APPROVED** [review]

**Pull request merged**